### PR TITLE
Us view pairs of interviewers

### PR DESCRIPTION
**Pull request by kmrozewski**
- add new resource to find pairs of interviewers with at least three slots per day
- find pairs in a specified dates range and optionally by a job profile(s)
- change availability times to enum
- unit tests
- add migrate changelog
- service for to get pair resources

### CODE REVIEW

File: src/main/java/domain/AvailabilityTypesEnum.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package domain;
+
+public enum AvailabilityTypesEnum {
```
Comment (wojciechh-kainos) [inline src/main/java/domain/AvailabilityTypesEnum.java:3]: You don't need to add Enum at the end of names here.

File: src/main/java/resources/PairResource.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    private SlotsDao slotsDao;
+
+    @Inject
+    public PairResource(SlotsDao slotsDao) {
+        this.slotsDao = slotsDao;
+    }
+
+    @GET
+    @Path("/find")
```
Comment (wojciechh-kainos) [inline src/main/java/resources/PairResource.java]: Why do we need a /find path? Can't we just do a GET to /pairs?

File: src/main/java/domain/AvailabilityTypesEnum.java (hunk 1 above)
Comment (msolecka-kainos) [inline src/main/java/domain/AvailabilityTypesEnum.java:3]: AvailabilityTypes already exists(in database). Could we rename it to AvailabilityTypesValue?

File: src/main/resources/assets/js/application/recruiter/services/tdprRecruiterViewPairsOfInterviewersService.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                }
+            }
+
+            return pathParams;
+        }
+
+        service.getPairs = function(roles, startDay, endDay){
+            var pathParams = this.createPathParams(roles, startDay, endDay);
+            if(!pathParams)
+                return false;
```
Comment (jraniszewski) [inline src/main/resources/assets/js/application/recruiter/services/tdprRecruiterViewPairsOfInterviewersService.js:28]: Missing brackets.

File: src/test/java/resourceTests/PairResourceTests/PairRecourceFindPairsForDifferentAvailabilityTypes.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+
+public class PairRecourceFindPairsForDifferentAvailabilityTypes {
```
Comment (mateuszs-kainos) [inline src/test/java/resourceTests/PairResourceTests/PairRecourceFindPairsForDifferentAvailabilityTypes.java]: A little mistake in class name "Recource"

File: src/main/java/resources/PairResource.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
+package resources;
```
Comment (mateuszs-kainos) [inline src/main/java/resources/PairResource.java:1]: In my opinion algorithm would be more understandable with a few comments.

File: src/main/resources/changelogs/pairsPopulateSlotsTable.xml [hunk 6]
```diff
@@ -0,0 +1,156 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
+
+    <changeSet id="15" author="konrad">
```
Comment (r-kruk-kainos) [inline src/main/resources/changelogs/pairsPopulateSlotsTable.xml]: Maybe change id for something different than just number.

File: src/test/java/resourceTests/PairResourceTests/PairResourcesFindPairsBetweenThreePersonsTest.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PairResourcesFindPairsBetweenThreePersonsTest {
```
Comment (mateuszs-kainos) [inline src/test/java/resourceTests/PairResourceTests/PairResourcesFindPairsBetweenThreePersonsTest.java]: A little mistake in class name "PairResource"

File: src/test/java/resourceTests/PairResourceTests/PairRecourceFindPairsForDifferentAvailabilityTypes.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        calendar.add(Calendar.DAY_OF_MONTH, 1);
+        Date nextDate = new Date(calendar.getTimeInMillis());
+
+        DateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+        startDate = dateFormat.format(date);
+        endDate = dateFormat.format(nextDate);
+
+        AvailabilityTypes typeAvailable = MockDataUtil.createAvailableType((long) 1, AvailabilityTypesEnum.available);
+        AvailabilityTypes typeMaybe = MockDataUtil.createAvailableType((long) 5, AvailabilityTypesEnum.maybe);
+
```
Comment (r-kruk-kainos) [inline src/test/java/resourceTests/PairResourceTests/PairRecourceFindPairsForDifferentAvailabilityTypes.java]: Is it possible to place test data in another file? This data seems similar in all test files.

File: src/test/java/resourceTests/PairResourceTests/PairResourcesFindPairsBetweenThreePersonsTest.java (hunk 7 above)
Comment (msolecka) [inline src/test/java/resourceTests/PairResourceTests/PairResourcesFindPairsBetweenThreePersonsTest.java]: Rename done! Thanks.

File: src/main/java/resources/PairResource.java [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+        this.slotsDao = slotsDao;
+    }
+
+    @GET
+    @UnitOfWork
+    public List<Pair> findPairs(@QueryParam("startDate") String startDate,
+                          @QueryParam("endDate") String endDate,
+                          @QueryParam("isDev") Boolean isDev,
+                          @QueryParam("isTest") Boolean isTest,
+                          @QueryParam("isOps") Boolean isOps) {
```
Comment (dawidk-kainos) [inline src/main/java/resources/PairResource.java]: isWeb instead of isOps. Let's be consistent.

**Pull request closed without merging**